File: stock/internal/service/stock_info.go
package service

import (
	"context"
	v1 "gostock/api/stock/v1"
	"gostock/internal/biz"
)

type StockService struct {
	v1.UnimplementedStockInfoServiceServer
	uc *biz.StockInfoUsecase
}

func NewStockInfoService(uc *biz.StockInfoUsecase) *StockService {
	return &StockService{
		uc: uc,
	}
}

func (s *StockService) GetStockInfo(ctx context.Context, in *v1.GetStockInfoRequest) (*v1.StockInfo, error) {
	g, err := s.uc.GetStockInfo(ctx, in.Id)
	if err != nil {
		return nil, err
	}
	stock := &v1.StockInfo{
		Id:   g.Id,
		Name: g.Name,
	}
	return stock, nil
}

File: crawler/crawler/internal/service/stock_info.py

from pykit.transport.http.context import Context
from api.crawler.v1.error_reason_pb2_erros import error_stock_not_found
from api.crawler.v1.stock_info_pb2 import StockBasicRequest, StockBasic
from api.crawler.v1.stock_info_pb2_grpc import StockServiceServicer
from crawler.internal.biz.stock_info import StockInfoUsecase


class StockInfoService(StockServiceServicer):
    def __init__(self, uc: StockInfoUsecase):
        self.uc = uc

    def GetStockInfo(self, req: StockBasicRequest, context: Context) -> StockBasic:

        res = StockBasic(ts_code='1', name='test')
        # raise error_stock_not_found(f"stock: {req.is_hs} not found")
        return res
